### ISSUE TITLE
How to keep the original size when running 'eval_vol'? 

### ISSUE THREAD

**Issue by minkyujeon** [alncat/opusDSD#12]
Hello, thank you for the excellent work! 

I've been attempting to apply k-means and generate volumes using 'eval_vol,' but the output volume size is downsampled due to cropping. 
Is there any way I could preserve the original image size when using 'eval_vol'? For instance, if I trained the data with an image size of 128, I would like to generate volumes with dimensions of 128x128x128.

Thanks!

**Comment by alncat** [comment]
Hi, sorry for the late reply! I am traveling this week so I didn't check this repo often. The dimension of volume generated by eval_vol is controlled by the ```apix``` argument, which is the fifth argument after ```eval_vol```. If you specify the original apix of the image, the volume reconstructed by eval_vol will have the same apix as the input image. But the dimension of volume might differ from the original size of image since opus-dsd will not output the redundant borders for a particle. You may notice that the cryo-EM images in the stack often have extra background borders without any densities for molecular structure. What really matters is the angstrom per pixel of the original image. The output volume reconstructed by ```eval_vol``` is scaled according the specified ```apix```, and will have the specified value.

**Comment by minkyujeon** [comment]
Hi @alncat, thanks for the reply! 
I want to compute the volume metric between the output of Opus-DSD and the ground truth volume (synthetic data). Therefore, I need to keep them the same size. 
Is there any way I could do to keep the size?

Additionally, I wonder if I could set --templateres to the default value (192) when the input image size is 128. 
Thanks a lot :)

**Comment by alncat** [comment]
@minkyujeon A simple way is specifing the original apix value when using eval_vol. You can then read the volumes reconstructed by OPUS-DSD into numpy arrays, and you can pad the reconstructed volume to the dimension of ground-truth volume using numpy or pytorch apis. Both sides should be padded. So the padding size should be (ground_truth_size - dsd_vol_size)/2 . 
Templateres 192 is big enough for the input image size 128. The templateres represents the volume output by the convnet. This volume is then resampled to the size of input image after masking (Redundant borders are removed). (Those regions outside mask are ignored since there is no densities for protein structure)

**Comment by alncat** [comment]
As long as you specify the orignal apix value, the angstrom per pixel of volume reconstructed by dsd is the same as the original ground-truths. The only difference is that the dimension of the volume reconstructed by dsd might be smaller than the original image since most of zero paddings are ignored by dsd.

**Comment by alncat** [comment]
@minkyujeon , I don't know if this trick still matters. I have a very easily way to pad volume to desire size, you can use relion_image_handler. For example, suppose you have a in.mrc with size 80x80x80, if you have relion 3.0.8, just do
```
relion_image_handler --i in.mrc --o out.mrc --new_box 120
```
then the out.mrc will be of size 120x120x120, and the in.mrc will be in the center box.

**Comment by RIA1159** [comment]
Hi there, I met the same problem here. When I use a synthetic data(size 128*128*128), the generate voluem is (84*84*84), and after I padding the volume with (ground_truth_size - dsd_vol_size)/2  to 128, the volume looks as large as the ground truth. However, the padding result(the green one) is not aligned with GT(the yellow one), and this non aligned result will affect the calculation of FSC indicators.

![Image](https://github.com/user-attachments/assets/7512d2e6-bd98-4f29-9b9b-e8cfab8b8348)

![Image](https://github.com/user-attachments/assets/aaf7a823-b768-4733-ab0b-5d2ed56e542e)

![Image](https://github.com/user-attachments/assets/19a1ca9a-8a6b-42ef-8cfd-46329da3c057)

**Comment by alncat** [comment]
Hi @RIA1159, Thank you very much for reporting this! If the volumes are not aligned, you can use ```fitmap #2 inMap #1``` to align them in chimerax. ```#2``` can be the volume generated by dsd, while ```#1``` is the reference volume.

**Comment by alncat** [comment]
@RIA1159 , then use ```volume resample #2 onGrid #1``` to create resampled volume, which should be ```#3```, you can then save the fitted volume using ```save fitted.mrc #3```.

**Comment by RIA1159** [comment]
Thanks ! That solved the problem!
